### PR TITLE
CLOUDP-59940: Upload dist files to s3

### PR DESCRIPTION
**Pull request by gssbzn**
## Proposed changes

_Jira ticket:_ [CLOUDP-59940](https://jira.mongodb.org/browse/CLOUDP-59940)

Sample run here https://evergreen.mongodb.com/task/mongocli_master_publish_package_patch_c8d11d9ea8854d3d4c79548e329bd7325043a6c3_5e83102cd6d80a16e2612d77_20_03_31_09_41_01

### CODE REVIEW

File: .evergreen.yml [hunk 1]
```diff
@@ -77,9 +91,40 @@ tasks:
...
+            set -o verbose
+
+            # we could generate a similar name with goreleaser but we want to keep the vars evg compatibale to use later
+            mv mongocli_next_linux_i386.deb mongocli_next_${revision}_${version_id}_${created_at}_linux_i386.deb
+            mv mongocli_next_linux_i386.rpm mongocli_next_${revision}_${version_id}_${created_at}_linux_i386.rpm
+            mv mongocli_next_linux_i386.tar.gz mongocli_next_${revision}_${version_id}_${created_at}_linux_i386.tar.gz
+            mv mongocli_next_linux_x86_64.deb mongocli_next_${revision}_${version_id}_${created_at}_linux_x86_64.deb
+            mv mongocli_next_linux_x86_64.rpm mongocli_next_${revision}_${version_id}_${created_at}_linux_x86_64.rpm
+            mv mongocli_next_linux_x86_64.tar.gz mongocli_next_${revision}_${version_id}_${created_at}_linux_x86_64.tar.gz
+            mv mongocli_next_macOS_x86_64.tar.gz mongocli_next_${revision}_${version_id}_${created_at}_macOS_x86_64.tar.gz
```
Comment (MihaiBojin) [inline .evergreen.yml]: nit: maybe make `macos` all lowercase, just like all the other output files

File: .evergreen.yml [hunk 2]
```diff
@@ -77,9 +91,40 @@ tasks:
...
       - func: "package"
+      - command: shell.exec
+        params:
+          working_dir: mongocli/dist
+          script: |
+            set -o errexit
+            set -o verbose
+
+            # we could generate a similar name with goreleaser but we want to keep the vars evg compatibale to use later
+            mv mongocli_next_linux_i386.deb mongocli_next_${revision}_${version_id}_${created_at}_linux_i386.deb
```
Comment (MihaiBojin) [inline .evergreen.yml]: I assume EVG populates `version_id` and `created_at`?

Comment (gssbzn) [reply .evergreen.yml]: yes, those are part of the default expansions, https://github.com/evergreen-ci/evergreen/wiki/Project-Files#expansions

File: .evergreen.yml (hunk 1 above)
Comment (gssbzn) [reply .evergreen.yml]: will do on a separate PR

**Pull request merged**